modules/mod_seo_sitemap/models/m_seo_sitemap.erl

```diff
@@ -0,0 +1,478 @@
+    Key = <<"seo_sitemap_rebuild_rsc">>,
+    {ok, _} = z_pivot_rsc:insert_task_after(10, ?MODULE, rebuild_rsc_task, Key, [ 1 ], Context),
+    ok.
+
+%% @doc Task for fetching the seo sitemap locs for the next 1000 ids. This task repeats
+%% till all ids are inserted into the seo_sitemap index table.
+-spec rebuild_rsc_task( m_rsc:resource_id(), z:context() ) -> ok | {delay, integer(), [ m_rsc:resource_id() ]}.
+rebuild_rsc_task(FromId, Context) ->
+    Ids = z_db:q("
+        select id from rsc where id >= $1
+        order by id asc
+        limit 1000
+        ", [ FromId ], Context),
+    Ids1 = [ Id || {Id} <- Ids ],
+    lists:foreach(
+        fun(Id) ->
+            update_rsc(Id, Context)
+        end,
+        Ids1),
+    case Ids1 of
+        [] -> ok;
+        _ -> {delay, 1, [ lists:last(Ids1)+1 ]}
+    end.
+
+%% @doc Insert entries for the given resource.
+-spec update_rsc( m_rsc:resource_id(), z:context() ) -> ok | {error, term()}.
+update_rsc(Id, Context) ->
+    AnonContext = z_context:new( z_context:site(Context) ),
+    case m_rsc:exists(Id, AnonContext)
+        andalso m_rsc:is_visible(Id, AnonContext)
+        andalso not z_convert:to_bool( m_rsc:p_no_acl(Id, seo_noindex, AnonContext) )
+    of
+        true ->
+            Langs = case m_rsc:p_no_acl(Id, language, Context) of
+                undefined -> [ z_context:language(AnonContext) ];
+                <<>> -> [ z_context:language(AnonContext) ];
+                [] -> [ z_context:language(AnonContext) ];
+                Ls -> Ls
+            end,
+            Enabled = editable_languages(AnonContext),
+            Locs = lists:filtermap(
+                fun(Lang) ->
+                    case lists:member(Lang, Enabled) of
+                        true ->
+                            CLang = z_context:set_language(Lang, AnonContext),
+                            Url = iolist_to_binary( m_rsc:p(Id, page_url, CLang) ),
+                            {true, {z_convert:to_binary(Lang), Url}};
+                        false ->
+                            false
+                    end
+                end,
+                Langs),
+            Locs1 = lists:usort(Locs),
+            Current = z_db:q("
+                        select language, loc
+                        from seo_sitemap
+                        where rsc_id = $1
+                          and source = 'rsc'",
+                        [ Id ],
+                        Context),
+            New = Locs1 -- Current,
+            Del = Current -- Locs1,
+            z_db:transaction(
+                fun(Ctx) ->
+                    lists:foreach(
+                        fun({_Lang, Loc}) ->
+                            z_db:q(
+                                "delete from seo_sitemap
+                                 where rsc_id = $1
+                                   and loc = $2
+                                   and source = 'rsc'",
+                                [ Id, Loc ],
```

Comment:
Shouldn't `lang` be included here too? Or included in the update on line 357?

Reply:
The `lang` is also assumed to be part of the url.